doc/governance/governance.rst

```diff
@@ -122,3 +122,78 @@ The following repositories are managed by the ros-controls PMC:
+It is very likely that ros2_control packages will be available via the ROS build farm earlier than this date.
+
+This stable release for a ROS distro, called **stable ros2_control release** from now, will be announced on `ROS discourse <https://discourse.openrobotics.org/c/ros-controls/ros-controls-announce-news/107>`__ on time.
+
+Versioning
+~~~~~~~~~~
+
+We will use the ROS-specific rules on top of ``semver's`` for versioning, but also adhere to some ros-controls-specific rules:
+
+* Major version increments (i.e. API breaking changes) should not be made within a **stable ros2_control release**.
+
+* ros2_control heavily relies on the usage of `pluginlib <https://index.ros.org/p/pluginlib/>`__. Therefore, we distinguish two types of compiled code: non-plugin code together with plugin base classes, and plugins itself.
+
+  * ABI of plugins may change at every release, i.e., also within a **stable ros2_control release**. Plugins built by the buildfarm will still be loaded by pluginlib's class loader, but code linking against the exported libraries will break.
+  * For non-plugin code, ABI breaks within a **stable ros2_control release** are less likely but still unavoidable to fix code which is critical for robot control.
+
+.. note::
+
+  * Always update your full ROS installation, not only a single package.
+  * Recompile your custom code after updating any upstream ROS packages.
+
+
+Public API declaration
+~~~~~~~~~~~~~~~~~~~~~~
+
+According to ``semver``, every package must clearly declare a public API.
+
+* For most C++ packages the declaration is any header that it installs. Private class members and methods are not part of the public API.
+
+* For other languages like Python, a public API must be explicitly defined, so that it is clear what symbols can be relied on with respect to the versioning guidelines.
+
+If something you are using is not explicitly listed as part of the public API in the package's documentation, then you cannot depend on it not changing between minor or patch versions.
+
+Deprecation strategy
+~~~~~~~~~~~~~~~~~~~~
+
+Where possible, we will use the tick-tock deprecation and migration strategy for breaking changes (API or behavior-breaking changes).
+
+* New deprecations can be run-time messages or compiler warnings expressing that the functionality is being deprecated. The functionality will be completely removed in any future release, or at latest in the next **stable ros2_control release** (there may be details in the deprecation note).
+
+* New deprecations can also come in every release of **stable ros2_control release** by performing backports of changes from the rolling version. These are meant to help users migrate early, however the functionality will remain available in that ROS distribution.
+
```

Comment:
```suggestion

The ros2_control maintainer team considers safety a top priority and bugs or issues found in the framework may be back ported to distros regardless of API stability.
These issues will be discussed at the monthly PMC meeting where the community can decide the best route forward.

```

Reply:
I think it would be a good idea to add a statement about bug fixes and back porting safety critical issues.